logger/logger.go

```diff
@@ -64,6 +64,10 @@ type logger struct {
 	Logger kitlog.Logger
 }
 
+func (l *logger) Decorate(keyvals ...interface{}) {
+	l.Logger = kitlog.NewContext(l.Logger).With(keyvals...)
```

Comment:
This could return new logger instance.
I think this should be possible. Otherwise why should we inject the logger to every service instead using global one?